### PR TITLE
Improve documentation.

### PR DESCRIPTION
**Pull request by ptersilie**
Co-authored-by: Edd Barrett <vext01@gmail.com>

### CODE REVIEW

File: internal_ws/ykcompile/src/stack_builder.rs [hunk 1]
```diff
@@ -1,3 +1,10 @@
+//! A stack builder for allocating objects on the stack.
+//!
+//! The builder is naive in that it allocates objects immediately (when `alloc` is called), thus
+//! potentially inserting more alignment padding than is necessary.
```
Comment (ltratt) [inline internal_ws/ykcompile/src/stack_builder.rs:4]: s/than is necessary/than would be necessary if we deferred and reordered allocations/ ?

File: internal_ws/ykpack/src/decode.rs [hunk 2]
```diff
@@ -1,9 +1,10 @@
+//! The pack decoder.
```
Comment (ltratt) [inline internal_ws/ykpack/src/decode.rs:1]: Missing blank line (I don't say that very often :p ).

File: internal_ws/ykshim/src/prod_api.rs [hunk 3]
```diff
@@ -1,3 +1,5 @@
+//! The ykshim API.
```
Comment (ltratt) [inline internal_ws/ykshim/src/prod_api.rs:1]: Can we expand this? It won't be very obvious to a new reader what `ykshim` is or what `prod` means (*I'm* only vaguely sure!).

File: internal_ws/yktrace/src/hwt/mapper.rs [hunk 4]
```diff
@@ -1,3 +1,6 @@
+//! The mapper loads SIR location labels from the executable and provides a way to map virtual
+//! block addresses to SIR locations, i.e. symbol name and block index.
```
Comment (ltratt) [inline internal_ws/yktrace/src/hwt/mapper.rs:2]: "i.e." or "e.g."?

File: ykrt/src/mt.rs [hunk 5]
```diff
@@ -1,3 +1,5 @@
+//! Initialise the meta-tracing system.
```
Comment (ltratt) [inline ykrt/src/mt.rs:1]: This one isn't exactly true. Maybe something like "The main end-user interface to the meta-tracing system"?

File: internal_ws/ykshim/src/prod_api.rs (hunk 3 above)
Comment (vext01) [reply internal_ws/ykshim/src/prod_api.rs:1]: `prod` is a term that will disappear. We agreed to merge `prod.rs` into `lib.rs`.

(Note there's already a comment in `lib.rs` which [explains what this is all about](https://github.com/softdevteam/yk/blob/8be961bb1e797253dd40a87a047de56cbd5c75d0/internal_ws/ykshim/src/lib.rs#L1-L4).)

Comment (vext01) [reply internal_ws/ykshim/src/prod_api.rs:1]: (I'll do this change as my next PR, so that it doesn't get forgotten)

File: ykrt/src/mt.rs (hunk 5 above)
Comment (ptersilie) [reply ykrt/src/mt.rs:1]: Hah, I just copied your suggestion from here: https://github.com/softdevteam/yk/pull/226/files#diff-9fa85f3ab0ab08c2bb4337dd17d47605f520c2780158e50e66eff7cd9dc1b29bR1

File: internal_ws/yktrace/src/hwt/mapper.rs (hunk 4 above)
Comment (ptersilie) [reply internal_ws/yktrace/src/hwt/mapper.rs:2]: "i.e." as in a SIR location is made up of a symbol name and block index.

**Comment by ptersilie** [comment]
This that addresses all comments.

File: internal_ws/ykshim/src/prod_api.rs (hunk 3 above)
Comment (ltratt) [reply internal_ws/ykshim/src/prod_api.rs:1]: Excellent!

**Comment by ltratt** [comment]
Please squash.

**Comment by ptersilie** [comment]
Squashed.

**Pull request merged**